src/esrs/postgres/event_store.rs

```diff
@@ -0,0 +1,297 @@
+use std::convert::TryInto;
+use std::sync::Arc;
+
+use arc_swap::ArcSwap;
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use futures::stream::BoxStream;
+use futures::StreamExt;
+use sqlx::pool::PoolConnection;
+use sqlx::postgres::{PgAdvisoryLock, PgAdvisoryLockGuard, PgAdvisoryLockKey};
+use sqlx::types::Json;
+use sqlx::{Executor, PgConnection, Pool, Postgres, Transaction};
+use uuid::Uuid;
+
+use crate::esrs::event_bus::EventBus;
+use crate::esrs::event_store::{EventStoreLockGuard, UnlockOnDrop};
+use crate::esrs::sql::event;
+use crate::esrs::sql::statements::{Statements, StatementsHandler};
+use crate::postgres::PgStoreError;
+use crate::types::SequenceNumber;
+use crate::{Aggregate, AggregateState, EventHandler, EventStore, StoreEvent, TransactionalEventHandler};
+
+/// Default Postgres implementation for the [`EventStore`]. Use this struct in order to have a
+/// pre-made implementation of an [`EventStore`] persisting on Postgres.
+///
+/// The store is protected by an [`Arc`] that allows it to be cloneable still having the same memory
+/// reference.
+#[derive(Clone)]
+pub struct PgStore<A>
```

Comment:
Why is the `PgStore` generic over the `Aggregate`?

If this was simply `PgStore` and then implement `EventStore<A>` we could have one singleton instance of the store that can be reused in the application. With the current version we need a specific store instance for each aggregate, which feels redundant (yes, I know the underlying `PgPool` is still shared, I'm talking about the convenience and developer experience of having multiple store instances around)

Reply:
But event handlers and transactional event handlers sets are unique per aggregate..

Reply:
Ah right... so as long as the store is in charge of the handlers it must be generic as well.

Bummers